addons/bus/websocket.py

```diff
@@ -47,6 +47,46 @@ def acquire_cursor(db):
     raise PoolError('Failed to acquire cursor after %s retries' % MAX_TRY_ON_POOL_ERROR)
 
 
+websocket_notifications = []
+websocket_events = []
+
+
+def log_websocket_stats():
+    while True:
+        time.sleep(60)
+        if websocket_notifications:
+            _logger.info(
+                "[GEVENT.DEBUG] Dispatched %d notifications to %d websockets in the last minute. Medium payload size: %d.",
+                len(websocket_notifications),
+                len({p[2] for p in websocket_notifications}),
+                sum(p[1] for p in websocket_notifications) // len(websocket_notifications),
+            )
+            count_by_type = {}
+            for p in websocket_notifications:
+                count_by_type.setdefault(p[0], 0)
+                count_by_type[p[0]] += 1
+            _logger.info(
+                "[GEVENT.DEBUG] Notification count by type: %s", json.dumps(count_by_type, indent=4)
+            )
+            websocket_notifications.clear()
+        if websocket_events:
+            _logger.info(
+                "[GEVENT.DEBUG] Received %d events from %d websockets in the last minute.",
+                len(websocket_events),
+                len({p[1] for p in websocket_events}),
+            )
+            count_by_event = {}
+            for p in websocket_events:
+                count_by_event.setdefault(p[0], 0)
+                count_by_event[p[0]] += 1
+            _logger.info(
+                "[GEVENT.DEBUG] Event count by type: %s", json.dumps(count_by_event, indent=4)
+            )
+            websocket_events.clear()
+
+
+threading.Thread(target=log_websocket_stats, daemon=True).start()
```

Comment:
This starts a thread as soon as this file is imported, no matter if bus is installed or not. In multi-worker it is going to start it in every http workers too. I suggest we move this call in a more appropriate place inside `service/server.py`.

Reply:
Sure, it's just a debug commit not meant to be merged :) Thanks for the suggestion.

Reply:
Oh sure, sorry I passed by your commit by watching which PR would be PR 177777 (funny number). Though it really was targeting 17.0 and stepped in. I'm fine with `if odoo.evented: threading...`, if it doesn't end up in real 17.0